extra_data/keydata.py

```diff
@@ -106,6 +106,28 @@ def size_gb(self):
         """The size of the data in memory in gigabytes."""
         return self.nbytes / 1e9
 
+    @property
+    def units(self):
+        dset = self.files[0].file[self.hdf5_data_path]
```

Comment:
Will this work with a voview file?

Reply:
Excellent point; no, at present it won't. I can add that to the code for creating virtual overview files, of course, but then it will only available on newly created files. :thinking:

Reply:
Reminds me of https://github.com/European-XFEL/DAMNIT/issues/93, except here we can't even assume that there's a `raw/` directory :sweat_smile: I think either having a method to follow the links in a voview file or adding the attributes to voview files is fine (I can regenerate all the existing files like I did before with the run metadata bug), but I lean towards having a method to follow the links. That seems a bit more future proof and would also be useful for that DAMNIT bug.

Reply:
OK, I've added an `.attributes()` method returning a dict of everything we get from the HDF5 dataset, and if that is empty and its a virtual dataset, it looks up one of the source files.